src/controllers/AuthController.ts

```diff
@@ -0,0 +1,39 @@
+import express from "express";
+import { getToken } from "../services/AuthService";
+//import * as AuthService from "../services/AuthService";
+
+export class AuthController {
+
+}
+
+export const getLoginForm = async (req: express.Request, res: express.Response): Promise<void> => {
+    res.render('loginForm.html', { error: 'Invalid credentials' });
+}
+
+export const postLoginForm = async (req: express.Request, res: express.Response): Promise<void> => {
+    const { Username, Password } = req.body;
+
+    if (!Username) {
+        return res.render('loginForm.html', { error: 'Username required'});
+    }
+    if (!Password){
+        return res.render('loginForm.html', { error: 'Password required'});
+    }
```

Comment:
you could update the html to set these fields as required to avoid needing this logic

Reply:
added required to html